unit-tests/live/d400/test-hdr.py

```diff
@@ -0,0 +1,575 @@
+
+        second_exposure = 1200
+        second_gain = 20
+        depth_sensor.set_option(rs.option.sequence_id, 2)
+        test.check(depth_sensor.get_option(rs.option.sequence_id) == 2)
+        depth_sensor.set_option(rs.option.exposure, second_exposure)
+        test.check(depth_sensor.get_option(rs.option.exposure) == second_exposure)
+        depth_sensor.set_option(rs.option.gain, second_gain)
+        test.check(depth_sensor.get_option(rs.option.gain) == second_gain)
+
+        depth_sensor.set_option(rs.option.hdr_enabled, 1)
+        test.check(depth_sensor.get_option(rs.option.hdr_enabled) == 1)
+
+        cfg = rs.config()
+        cfg.enable_stream(rs.stream.depth)
+        cfg.enable_stream(rs.stream.infrared, 1)
+        pipe = rs.pipeline()
+        pipe.start(cfg)
+
+        for iteration in range(1, 100):
+            data = pipe.wait_for_frames()
+
+            out_depth_frame = data.get_depth_frame()
+            if iteration < 3:
+                continue
+
+            if out_depth_frame.supports_frame_metadata(rs.frame_metadata_value.sequence_id):
+                frame_exposure = out_depth_frame.get_frame_metadata(rs.frame_metadata_value.actual_exposure)
+                frame_gain = out_depth_frame.get_frame_metadata(rs.frame_metadata_value.gain_level)
+                seq_id = out_depth_frame.get_frame_metadata(rs.frame_metadata_value.sequence_id)
+
+                if seq_id == 0:
+                    test.check(frame_exposure == first_exposure)
+                    test.check(frame_gain == first_gain)
+                else:
+                    test.check(frame_exposure == second_exposure)
+                    test.check(frame_gain == second_gain)
+
+        pipe.stop()
+        depth_sensor.set_option(rs.option.hdr_enabled, 0)  # disable hdr before next tests
+        device.hardware_reset()
+        time.sleep(5)
+
+
+# CHECKING HDR AFTER PIPE RESTART
+with test.closure("HDR Running - restart hdr at restream"):
+    device = test.find_first_device_or_exit()
+    depth_sensor = device.first_depth_sensor()
+    if depth_sensor and depth_sensor.supports(rs.option.hdr_enabled):
+        cfg = rs.config()
+        cfg.enable_stream(rs.stream.depth)
+        pipe = rs.pipeline()
+        pipe.start(cfg)
+        depth_sensor.set_option(rs.option.hdr_enabled, 1)
+        test.check(depth_sensor.get_option(rs.option.hdr_enabled) == 1)
+        for i in range(10):
+            data = pipe.wait_for_frames()
+        test.check(depth_sensor.get_option(rs.option.hdr_enabled) == 1)
+        pipe.stop()
+        pipe.start(cfg)
+        test.check(depth_sensor.get_option(rs.option.hdr_enabled) == 1)
+        pipe.stop()
+        depth_sensor.set_option(rs.option.hdr_enabled, 0)  # disable hdr before next tests
+
+
+"""
+helper method
+checks that the frames resulting from the hdr_merge processing block are generated
+from frames streamed within this method (and not using old frames) - see RSDSO-17341.
```

Comment:
I would change "within this method" to "within this streaming session" - only if you agree